tsconfig.base.json

```diff
@@ -1,21 +1,34 @@
 {
   "compilerOptions": {
+    "allowJs": false,
     "allowSyntheticDefaultImports": true,
     "allowUnreachableCode": false,
     "allowUnusedLabels": false,
+    "composite": true,
     "declaration": true,
     "declarationMap": true,
+    "emitDeclarationOnly": true,
+    "emitDecoratorMetadata": false,
     "esModuleInterop": true,
-    "lib": ["ES2021"],
-    "module": "Node16",
-    "moduleResolution": "Node16",
+    "experimentalDecorators": false,
+    "forceConsistentCasingInFileNames": true,
```

Comment:
A lot of these settings are already their defaults and could be removed for clarity, e.g. allowJs, decorators, pretty, and so on.

Reply:
Yeah I think that's probably fair... I was shooting for maximum clarity, but probably the fewer things for the average user to think about the better. Y'all are striving for the most logical/useful default values and on the whole users don't need to care about many of the possible options